.github/workflows/tests-e2e-npm.yml

```diff
@@ -37,16 +41,22 @@ jobs:
         run: |
           echo "Parsing JSON environment variables..."
           echo "$ENV_VARS_JSON" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]' >> $GITHUB_ENV
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.node_version }}
+        if: inputs.docker_compose_file == ''
```

Comment:
we always want to setup node don't we?

Reply:
if we would use docker-compose file I don't think we'd need it ...but yes if we always use testcontainers then we'll always want to set it up

Reply:
We need it if we run the tests with `npm`. Given this is the `tests-e2e-npm` workflow that is almost certainly a given I think